client/src/pages/SignIn.js

```diff
@@ -64,12 +65,41 @@ const signInPageStyle = (theme) => ({
   },
 });
 
+
 const SignInPage = (props) => {
   const { classes } = props;
+  let history = useHistory();
+  const { user, setUser } = useContext(UserContext);
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
+  const [message, setMessage] = useState(null);
+
+  const handleSignIn = (e) => {
+    e.preventDefault();
+    const data = {email: email, password: password};
+    fetch('/auth/login', {
+      method: "POST",
+      headers: {
+        "Content-type": "application/json"
+      },
+      redirect: "manual",
+      body: JSON.stringify(data)
+    }).then(response => response.json()).then(data => {
+      if (data.status === 200) {
+        console.log(user);
+        setUser(data.user);
+        console.log(data.user.profile);
+        if (data.user.profile.is_chef){
+          history.push('/chef_profile')
 
-  console.log(email, password); // Just added for now to avoid the warning saying both are assigned, but not used.
+        }else{
+        history.push('/user_profile')
+        }
```

Comment:
Let's stick consistently to a navigation style. If we're gonna use React-Router-Dom's React components, let's stick to those instead of mixing them with `history.push`. So the way you could handle this is to have a check to see if the user is logged in in `render()`, then if so, based on `user.profile.isChef`, render `<Redirect to="" />` the appropriate page

Reply:
I tried using <Redirect> but it refused to do so and every time I tried using render() it would tell me it expected a ; or , 
I've tried using redirect: "follow" in the fetch, tried returning the UserProfile component directly, tried wrapping it in UserContext.Provider value={user} and returning that, Tried returning <Redirect/> directly, tried wrapping that in UserContext.provider, tried using render and it would complain about expecting a semicolon or comma but nothing worked and I couldn't find anything online. Everything said to use history.push so I tried and it worked. Dan also said it was a good solution so I figured it would be fine.
This was the only thing that worked. I'm not sure how to do what you're asking because, like I said, none of it worked. Not sure if it's because it's in a nested function? I read that fetch doesn't follow redirects but I'm almost positive that means redirects sent from the backend. Any advice would be greatly appreciated.